kubespawner/proxy.py

```diff
@@ -151,6 +121,28 @@ class is Proxy, as declared in proxy.py:
         self.endpoint_reflector = EndpointsReflector(
             self, namespace=self.namespace, labels=labels
         )
+
+        # Schedules async initialization logic that is to be awaited by async
+        # functions by decorating them with @_await_async_init.
+        # this is a single Future representing all reflectors having loaded at last once
+        self._reflectors_ready = asyncio.ensure_future(self._async_init())
+
+    async def _async_init(self):
+        """
+        This method is scheduled to run from `__init__`, but not awaited there
+        as `__init__` can't be marked as async.
+
+        Since JupyterHub won't await this method, we ensure the async methods
+        that may be called on this object will await this method before
+        continuing.
+        To do this, we can decorate them with `@_await_async_init`.
+
+        But, how do we figure out the methods to decorate?
+        Any method that accesses the state created in this method.
+
+        Currently, that is only the first-load of our reflectors,
+        which can also be awaited individually with `await reflector._first_load_future`.
```

Comment:
The comments here seems out of date with the rest of the changes

Reply:
Updated to be shorter and hopefully be more precise.